### PR TITLE
Add DSB instruction to AArch64 opcode tables

### PR DESCRIPTION
**Pull request by hsoontie**
This PR adds DSB to the opcode tables in AArch64, as it is
needed in OpenJ9.

Signed-off-by: Hillary Soontiens <hsoontie@unb.ca>

### CODE REVIEW

**Comment by hsoontie** [comment]
@0xdaryl, here is this, to go along with #6250 in openj9.

**Comment by 0xdaryl** [comment]
The encoding changes look fine.

However, one thing I just realized is that an `Imm` form instruction is inadequate to encode these instructions because the "immediates" are different sizes and they are in a different bit position.   Sorry I didn't clue into this sooner.

 I think the best way to solve this is to introduce a new instruction kind (e.g., an `AMD64SynchronizationInstruction` kind) that accepts an immediate and encodes the 4-bit field as the synchronization instructions expect.  You could also add some checking (via a `TR_ASSERT`) if someone attempted to specify a value that can't be represented in 4-bits.  This would apply to the `DMB` family of instructions as well should we need to support that at some point in the future.

**Comment by 0xdaryl** [comment]
Could you add the new instruction kind as a separate commit in this PR please?

**Comment by hsoontie** [comment]
@0xdaryl, yes, I was wondering whether or not using `Imm` would work. I will start on this as soon as I can.

**Comment by hsoontie** [comment]
@0xdaryl Here is a new commit with my implementation of `ARM64SynchronizationInstruction`. Please let me know your thoughts when you have a moment. Thanks!

File: compiler/aarch64/codegen/ARM64Instruction.hpp [hunk 1]
```diff
@@ -2629,6 +2629,72 @@ class ARM64Src2Instruction : public ARM64Src1Instruction
...
+   /**
+    * @brief Sets source immediate
+    * @param[in] si : immediate value
+    * @return source immediate
+    */
+   uint32_t setSourceImmediate(uint32_t si) {return (_sourceImmediate = si);}
+
+   void insertImmediateField(uint32_t *instruction)
+      {
+      TR_ASSERT(_sourceImmediate < 0xF, "Immediate value exceeds 4 bits.");
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/ARM64Instruction.hpp]: Operator should be `<=` for 4-bits.

File: compiler/aarch64/codegen/ARM64Instruction.hpp [hunk 2]
```diff
@@ -2629,6 +2629,72 @@ class ARM64Src2Instruction : public ARM64Src1Instruction
...
+      *instruction |= (_sourceImmediate << 8);
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/ARM64Instruction.hpp]: Just to be on the safe size, can you mask just the lower 4 bits?  i.e.,
```
 *instruction |= ((_sourceImmediate & 0xF) << 8);
```

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

I'll just do a build to see if there are any compile issues.

**Comment by hsoontie** [comment]
@0xdaryl I am not sure what that build failure means.
As for the changes, I'll take care of them asap.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

Let's try once more.  Looks like an infra issue.

**Comment by 0xdaryl** [comment]
The build passed now.

**Comment by hsoontie** [comment]
@0xdaryl Gave this an update, and it looks like everything tested fine.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

**Pull request merged**